### DISCUSSION TITLE
ARM64 vmlinuz in CentOS 7/8 official qcow2 can't be select in  iPXE EFI mode

### DISCUSSION THREAD

**Discussion by C2H5OHlife** [ipxe/ipxe#185] [General]
Hi, I'm trying to pair an aarch64 kernel with an initrd to boot in-memory OS using iPXE on an ARM machine.

I made the `snponly.efi` binary via command `make bin-arm64-efi/snponly.efi` and chainloding it seems good.

For getting the kernel, I extracted the vmlinuz file from both the offical CentOS 7/8 qcow2 images in path `./boot/vmlinuz-${version}.aarch64` and used the iPXE command line to handle them :
```
kernel http://.../vmlinuz-${version}.aarch64
```
They can be fetched properly, but neither of them can be be selected:
```
http://.../vmlimuz-${version}.aarch64 ... ok
Could not select: Exec format error (http://ipxe.org/2e008001)
```
I visit that link and check the `CONFIG_EFI_STUB` option in the config files `./boot/config-${version}.aarch64` which I believe are the config used in the make of those kernels. And all of them have been already set:
```
CONFIG_EFI_STUB=y
CONFIG_EFI=y
```
So any ideas to solve this error? Or have I did anything wrong in my operations?

**Comment by NiKiZe** [comment]
What iPXE does here is simply to tell the firmware to run that binary, iso it all depends on the firmware, not iPXE.

You could try an efi shell and run the binary from there.

One thing you can do from iPXE is to load the kernel, then run imgstat to see that it is good, for example that the server has responded with the full file and not a partial file or error message.
